### ISSUE TITLE
Multiple tags and HAProxy

### ISSUE THREAD

**Issue by thomas-oo** [hashicorp/consul-template#953]
This is semi related to this [issue here](https://github.com/hashicorp/consul-template/issues/33).
I am using HAProxy as a proxy service to route a request to a particular service. This is working well for routing per service name ie: REST service, blog service etc. All I write in the template is write out a new backend for every service and list the service nodes under it.

Now my use case becomes more complicated with support for querying with tags. Say I want a REST service but with tags: active, version2, etc. I now need to know of which REST service, also has these tags.

Should I therefore, make the template write out the backends with combinations of names and tags, and then use ACL to route the request to that backend? Do you guys have suggestions on how to tackle this problem?

TLDR: I need help to support routing to a specific backend based on service name, and its tag such that the backend it gets routed to, its nodes match the tags and service name.

**Comment by thomas-oo** [comment]
I just read that consul upstream only allows filtering by one tag anyways. If anyone was looking to accomplish a similar use case, I suggest putting a seperate tag that handles the conjuction of multiple tags. Eg have a tag "active", "version2", but also have a tag "activeANDversion2". Then when you tag a service as "activeANDversion2", also tag it as "active", "version2". This also makes it easier for you to create seperate backends for seperate tags.

Let me know if any alternatives exist!
